### PR TITLE
feat(api) : validation message refactored

### PR DESCRIPTION
**Pull request by soumyadip007**
Subject:  feat(api) : validation message refactored
Assignees: @soumyadip007 

## Closes / Fixes 

1.  Validations message refactored for the Application Deploy response
2. CLI will show these message in Error (rather than generic error)
3. Identifier added for activity stream

## Does this PR introduce a breaking change

NO

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/api/src/configuration/index.ts [hunk 1]
```diff
@@ -45,24 +45,24 @@ export enum MAX {
...
   CLUSTER = 12
 }
 
 export enum MESSAGE {
-  INVALID_LENGTH = 'Length is Invalid.',
-  INVALID_PROPERTY_TITLE = 'Invalid Property Title.',
-  INVALID_PROPERTY_IDENTIFIER = 'Invalid Property Identifier.',
+  INVALID_LENGTH = 'Length is Invalid',
+  INVALID_PROPERTY_TITLE = 'Invalid Property Title [Example : Test Property].',
+  INVALID_PROPERTY_IDENTIFIER = 'Invalid Property Identifier [Example : test-property].',
```
Comment (hybridx) [inline packages/api/src/configuration/index.ts:55]: @soumyadip007 why are we mentioning test property?

File: packages/api/src/configuration/index.ts [hunk 2]
```diff
@@ -45,24 +45,24 @@ export enum MAX {
...
-  INVALID_ACTION_ID = 'Invalid Action Id [Please provide the Pull Request ID, Ex: 101].',
-  INVALID_APPLICATION = 'Invalid Application name.',
-  INVALID_REF = 'Invalid Reference.',
-  INVALID_ENV = 'Invalid Environment',
-  INVALID_LABEL = 'Invalid Label.'
+  INVALID_ACTION_ID = 'Invalid Action Id [Please provide the Pull Request ID, Example: 101].',
+  INVALID_APPLICATION = 'Invalid Application name [Example : home, test, redhat doc].',
+  INVALID_REF = 'Invalid Reference [Example : 1.2, v1@1.2.3].',
+  INVALID_ENV = 'Invalid Environment [Example : prod, stage, dev]',
+  INVALID_LABEL = 'Invalid Label [Example : all-access, prod-key].'
```
Comment (hybridx) [inline packages/api/src/configuration/index.ts:65]: I think we can come up with better error response than this

File: packages/api/src/configuration/index.ts [hunk 3]
```diff
@@ -76,7 +76,7 @@ export const VALIDATION = {
   LABEL: /^[_a-zA-Z0-9 -.]+$/,
   URL: /^[_a-zA-Z0-9-.]+$/,
   ENV: /^[a-zA-Z0-9-]+$/,
-  REF: /^[_a-zA-Z0-9/ -.]+$/,
+  REF: /^[_a-zA-Z0-9/@ -.]+$/,
```
Comment (hybridx) [inline packages/api/src/configuration/index.ts:79]: Is this for the new fix?

**Review by hybridx — CHANGES_REQUESTED** [review]

File: packages/api/src/configuration/index.ts (hunk 3 above)
Comment (soumyadip007) [reply packages/api/src/configuration/index.ts:79]: Yes for allowing @

File: packages/api/src/configuration/index.ts (hunk 1 above)
Comment (soumyadip007) [reply packages/api/src/configuration/index.ts:55]: As an example we are writing this.

File: packages/api/src/configuration/index.ts (hunk 2 above)
Comment (soumyadip007) [reply packages/api/src/configuration/index.ts:65]: Any better format ?

**Review by hybridx — APPROVED** [review]

**Pull request merged**